exec/jdbc/src/main/java/org/apache/drill/jdbc/impl/DrillResultSetImpl.java

```diff
@@ -87,12 +88,12 @@
   boolean hasPendingCancelationNotification;
 
 
-  DrillResultSetImpl(DrillStatementImpl statement, AvaticaPrepareResult prepareResult,
+  DrillResultSetImpl(AvaticaStatement statement, AvaticaPrepareResult prepareResult,
                      ResultSetMetaData resultSetMetaData, TimeZone timeZone) {
     super(statement, prepareResult, resultSetMetaData, timeZone);
-    this.statement = statement;
+    connection = (DrillConnectionImpl) statement.getConnection();
     final int batchQueueThrottlingThreshold =
-        this.getStatement().getConnection().getClient().getConfig().getInt(
+        connection.getClient().getConfig().getInt(
```

Comment:
if you move the line `this.client = client;` before this line you should be able to reuse `this.client` instead of calling `connection.getClient()`
You can also reuse `this.connection` instead of creating another instance `c` below